app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +109,69 @@ class SettingsActivity : AppCompatActivity() {
             .commit()
     }
 
+    override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
+        (supportFragmentManager.findFragmentById(R.id.settings) as PreferenceFragmentCompat).preferenceScreen.forEach { preferenceCategory ->
+            var noKey = true
+            (preferenceCategory as PreferenceCategory).forEach { preference ->
+                if (preference.key != null) {
+                    mapKeyTitle[preference.title.toString()] = preference
+                    noKey = false
+                }
+            }
+            if (!noKey)
+                mapKeyTitle[preferenceCategory.title.toString()] = preferenceCategory
+        }
+
+        menuInflater.inflate(R.menu.menu, menu)
+        val menuItem = menu!!.findItem(R.id.app_bar_search)
+        val searchView = menuItem.actionView as SearchView
+        searchView.queryHint = getString(R.string.search)
+
+        searchView.setOnQueryTextFocusChangeListener { _, focus ->
+            (binding.titlebar.toolbar.layoutParams as AppBarLayout.LayoutParams).scrollFlags =
+                if (focus)
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_NO_SCROLL
+                else
+                    AppBarLayout.LayoutParams.SCROLL_FLAG_SCROLL
+        }
+
+        searchView.setOnQueryTextListener(object : SearchView.OnQueryTextListener {
+            override fun onQueryTextSubmit(query : String?) : Boolean {
+                return false
+            }
+
+            override fun onQueryTextChange(newText : String?) : Boolean {
+                val querys = newText!!.split(",").toTypedArray()
```

Comment:
If `newText` is null you should just return here, and not crash. Look if you can make `newText` non-nullable too.